lahja/endpoint.py

```diff
@@ -266,35 +238,49 @@ class Endpoint(BaseEndpoint):
     The :class:`~lahja.endpoint.Endpoint` enables communication between different processes
     as well as within a single process via various event-driven APIs.
     """
-
-    _name: str
-    _ipc_path: pathlib.Path
+    __slots__ = (
+        'ipc_path',
+        '_child_tasks',
+        '_full_connections',
+        '_futures',
+        '_half_connections',
+        '_handlers',
+        '_internal_queue',
+        '_queues',
+        '_receiving_loop_running',
+        '_receiving_queue',
+        '_receiving_queue',
+        '_running',
+        '_server_running',
+        '_serving',
+        '_who_are_you_stream_handler_running',
+    )
+
+    ipc_path: pathlib.Path
 
     _receiving_queue: 'asyncio.Queue[Tuple[Union[bytes, BaseEvent], Optional[BroadcastConfig]]]'
     _receiving_loop_running: asyncio.Event
 
-    _internal_queue: 'asyncio.Queue[Tuple[BaseEvent, Optional[BroadcastConfig]]]'
-    _internal_loop_running: asyncio.Event
-
     _server_running: asyncio.Event
 
-    _loop: Optional[asyncio.AbstractEventLoop]
+    _loop: Optional[asyncio.AbstractEventLoop] = None
+
+    def __init__(self, name: str) -> None:
+        self.name = name
```

Comment:
@cburgdorf thoughts on this being core API? Core API is that endpoints **always** have a name.